pkg/volume/cephfs/cephfs_test.go

```diff
@@ -215,13 +215,63 @@ func TestGetSecretNameAndNamespaceForPV(t *testing.T) {
 			expectedNs:    "default",
 			expectedError: nil,
 		},
+		{
+			name:      "pod volume source no secretRef",
+			defaultNs: "default",
+			spec: &volume.Spec{
+				Volume: &v1.Volume{
+					VolumeSource: v1.VolumeSource{
+						CephFS: &v1.CephFSVolumeSource{
+							Monitors: []string{"a", "b"},
+							User:     "user",
+						},
+					},
+				},
+			},
+			expectedName:  "",
+			expectedNs:    "",
+			expectedError: nil,
+		},
+		{
+			name:          "no volume",
+			spec:          &volume.Spec{},
+			expectedName:  "",
+			expectedNs:    "",
+			expectedError: fmt.Errorf("Spec does not reference an CephFS volume type"),
+		},
 	}
 	for _, testcase := range tests {
 		resultName, resultNs, err := getSecretNameAndNamespace(testcase.spec, testcase.defaultNs)
-		if err != testcase.expectedError || resultName != testcase.expectedName || resultNs != testcase.expectedNs {
-			t.Errorf("%s failed: expected err=%v ns=%q name=%q, got %v/%q/%q", testcase.name, testcase.expectedError, testcase.expectedNs, testcase.expectedName,
-				err, resultNs, resultName)
+		switch testcase.name {
+		case "no volume":
```

Comment:
Why do you need to special case this testcase to match the error?

Reply:
because 
logs.Info("err and err.Error() ", reflect.TypeOf(err), reflect.TypeOf(err.Error()))
err and err.Error()  *errors.errorString string
the err if not nil can not compare ,err is errorString struct， if compare must convert err.Error()  to string

Reply:
Can you change it for all testcases then?  Why should only the "no volume" testcase have this special check?

Reply:
thanks,I will research it,I test it but make failed! then modify the follow func,for your review again,really thanks!